Assignment/Book-Management/src/app/core/services/book.service.ts

```diff
@@ -0,0 +1,49 @@
+import { Router } from '@angular/router';
+import { HttpClient, HttpErrorResponse, HttpHeaders } from "@angular/common/http";
+import { Injectable, ElementRef } from '@angular/core';
+import { Observable, from } from 'rxjs';
+import { tap } from'rxjs/operators';
+
+// export interface BookResponseData {
+//   idToken:	string,
+//   email: string,
+//   refreshToken: string,
+//   expiresIn: string,
+//   localId: string,
+//   register?: boolean
+// }
+
+
+@Injectable({ providedIn: 'root'})
+export class BookService {
+  constructor(private http: HttpClient, private route: Router) {}
+  getObs: Observable<any>;
+  totalPages: number;
+  totalItems: number;
+
+  getAllBook(pageIndex: number, pageSize: number) {
+
+    this.http.get<any>('https://nga-book-api.herokuapp.com/api/books?pageNumber='+pageIndex+'&pageSize='+pageSize, {
```

Comment:
- Store https://nga-book-api.herokuapp.com/api in environment.ts.
- Use template literals. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

Reply:
done